src/plugin.js

```diff
@@ -269,17 +267,27 @@ class VR extends Plugin {
   }
 
   handleVrDisplayActivate_() {
-    this.manager.enterVRMode_();
-    this.manager.setMode_(3);
+    if (!this.vrDisplay) {
+      return;
+    }
+    this.vrDisplay.requestPresent([{source: this.renderedCanvas}]);
   }
 
   handleVrDisplayDeactivate_() {
-    if (!this.vrDisplay.isPresenting) {
+    if (!this.vrDisplay || !this.vrDisplay.isPresenting) {
       return;
     }
     this.vrDisplay.exitPresent();
   }
 
+  requestAnimationFrame(fn) {
+    if (this.vrDisplay) {
```

Comment:
If we have a vr display we want to request animation from on that instead of the plugin.